test/integration/appfactory/appfactory_test.go

```diff
@@ -48,179 +48,183 @@ func TestAppfactory(t *testing.T) {
-					listMembers := utils.GetResultStrSlice(repoIamPolicyOp.Get("bindings.members").Array())
-					assert.Contains(listMembers, repoSa, fmt.Sprintf("Service Account %s should have role roles/viewer on repo %s", repoSa, repositoryName))
-
-					// check cloudbuild workspace
-					// buckets
-					gcloudArgsBucket := gcloud.WithCommonArgs([]string{"--project", adminProjectID, "--json"})
-					for _, bucket := range []struct {
-						output string
-						suffix string
-						prefix string
-					}{
-						{
-							output: "app_cloudbuild_workspace_state_bucket_name",
-							suffix: "state",
-							prefix: "bkt",
-						},
-						{
-							output: "app_cloudbuild_workspace_logs_bucket_name",
-							suffix: "logs",
-							prefix: "bkt",
-						},
-						{
-							output: "app_cloudbuild_workspace_artifacts_bucket_name",
-							suffix: "build",
-							prefix: "bkt",
-						},
-					} {
-						bucketSelfLink := appData.Get(bucket.output).String()
-						opBucket := gcloud.Run(t, fmt.Sprintf("storage ls --buckets gs://%s-%s-%s-%s", bucket.prefix, adminProjectID, serviceName, bucket.suffix), gcloudArgsBucket).Array()
-						assert.Equal(bucketSelfLink, opBucket[0].Get("metadata.selfLink").String(), fmt.Sprintf("The bucket SelfLink should be %s.", bucketSelfLink))
-					}
-					// triggers
-					repoName := appData.Get("app_infra_repository_name").String()
-					for _, trigger := range []struct {
-						output string
-						file   string
-					}{
-						{
-							output: "app_cloudbuild_workspace_apply_trigger_id",
-							file:   "cloudbuild-tf-apply.yaml",
-						},
-						{
-							output: "app_cloudbuild_workspace_plan_trigger_id",
-							file:   "cloudbuild-tf-plan.yaml",
-						},
-					} {
-						triggerID := testutils.GetLastSplitElement(appData.Get(trigger.output).String(), "/")
-						buildTrigger := gcloud.Runf(t, "builds triggers describe %s --project %s --region %s", triggerID, adminProjectID, "global")
-						filename := buildTrigger.Get("filename").String()
-						assert.Equal(trigger.file, filename, fmt.Sprintf("The filename for the trigger should be %s but got %s.", trigger.file, filename))
-						assert.Equal(repoName, buildTrigger.Get("triggerTemplate.repoName").String(), "the trigger should use the repo %s", repoName)
-					}
+					})
+				}
 
-					if slices.Contains(testutils.ServicesWithEnvProject[appName], serviceName) {
-						// check env projects
-						cloudBuildSARoles := []string{"roles/owner"}
-						envProjectsIDs := appData.Get("app_env_project_ids")
-						envProjectApis := []string{
+				// check admin projects
+				// TODO: Update to use https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/2356 when released.
+				// terraform.OutputJson OK to use as long as there is only one appGroupName
+				for applicationService, appData := range gjson.Parse(terraform.OutputJson(t, appFactory.GetTFOptions(), "app-group")).Map() {
```

Comment:
I suggest using `appFactory.GetJsonOutput()` as that includes mutex locking for the TF provider cache for parallel (tf init) operations.